src/bgw/scheduler.c

```diff
@@ -229,8 +229,6 @@ worker_state_cleanup(ScheduledBgwJob *sjob)
 			 */
 			elog(LOG, "job %d failed", sjob->job.fd.id);
 			mark_job_as_ended(sjob, JOB_FAILURE);
-			/* reload updated value */
-			job_stat = ts_bgw_job_stat_find(sjob->job.fd.id);
 		}
```

Comment:
What side-effects exactly? It looks like it's just a catalog lookup. I'm not sure how to best test this change, maybe I should remove it in a follow-up PR. I think it's a remnant of dead store detection which I ultimately disabled because it requires too many changes.